applications/tari_indexer/src/substate_storage_sqlite/models/events.rs

```diff
@@ -58,28 +63,62 @@ pub struct EventData {
     pub topic: String,
     #[diesel(sql_type = Text)]
     pub payload: String,
+    #[diesel(sql_type = Integer)]
+    pub version: i32,
+    #[diesel(sql_type = Nullable<Text>)]
+    pub component_address: Option<String>,
 }
 
 impl TryFrom<EventData> for crate::graphql::model::events::Event {
     type Error = anyhow::Error;
 
     fn try_from(event_data: EventData) -> Result<Self, Self::Error> {
-        let mut template_address = [0u8; 32];
-        let template_address_buff =
-            from_hex(event_data.template_address.as_ref()).map_err(|e| anyhow::anyhow!(e.to_string()))?;
-        template_address.copy_from_slice(&template_address_buff);
+        let component_address = event_data
+            .component_address
+            .map(|comp_addr| ComponentAddress::from_str(comp_addr.as_str()))
+            .transpose()?
+            .map(|comp_addr| comp_addr.into_array());
 
-        let mut tx_hash = [0u8; 32];
-        let tx_hash_buffer = from_hex(event_data.tx_hash.as_ref()).map_err(|e| anyhow::anyhow!(e.to_string()))?;
-        tx_hash.copy_from_slice(&tx_hash_buffer);
+        let template_address = Hash::from_hex(&event_data.template_address)
+            .map_err(|e| anyhow::anyhow!(e.to_string()))?
+            .into_array();
+
+        let tx_hash = Hash::from_hex(&event_data.tx_hash)
+            .map_err(|e| anyhow::anyhow!(e.to_string()))?
+            .into_array();
 
         let payload = serde_json::from_str(event_data.payload.as_str()).map_err(|e| anyhow::anyhow!(e.to_string()))?;
 
         Ok(Self {
+            component_address,
             template_address,
             tx_hash,
             payload,
             topic: event_data.topic,
         })
     }
 }
+
+impl TryFrom<EventData> for tari_engine_types::events::Event {
+    type Error = anyhow::Error;
+
+    fn try_from(event_data: EventData) -> Result<Self, Self::Error> {
+        let component_address = event_data
+            .component_address
+            .clone()
+            .map(|comp_addr| ComponentAddress::from_str(comp_addr.as_str()))
+            .transpose()?;
+        let template_address =
+            Hash::from_hex(&event_data.template_address).map_err(|e| anyhow::anyhow!(e.to_string()))?;
+        let tx_hash = Hash::from_hex(&event_data.tx_hash).map_err(|e| anyhow::anyhow!(e.to_string()))?;
+        let payload = serde_json::from_str(event_data.payload.as_str()).map_err(|e| anyhow::anyhow!(e.to_string()))?;
```

Comment:
Anyhow implements `From` for any `Error` and `?` desugars to
```rust
match stmt {
   Ok(v) => v,
   Err(e) => return Err(e.into()), // Note the .into()
}
```

```suggestion
        let template_address =
            Hash::from_hex(&event_data.template_address)?;
        let tx_hash = Hash::from_hex(&event_data.tx_hash)?;
        let payload = serde_json::from_str(event_data.payload.as_str())?;
```